Tests/Functional/FunctionalTestCase.php

```
@@ -0,0 +1,293 @@
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->nodeTypeManager = $this->objectManager->get(NodeTypeManager::class);
+        $this->currentUserWorkspace = explode('.', uniqid('user-', true))[0];
+        $this->currentGroupWorkspace = explode('.', uniqid('group-', true))[0];
+        $this->setUpRootNodeAndRepository();
+
+        // Purge existing sites/domains and content under /sites to ensure tests are isolated
+        $this->purgeSitesDomainsAndContent();
+
+        foreach ($this->siteHosts as $i => $siteHost) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$i'. (undefined)

(UnusedLocalVariable)